test/clojure/data/csv_test.clj

```diff
@@ -78,15 +81,42 @@ air, moon roof, loaded\",4799.00")
     (is (= ["Year" "Make" "Model"] (first csv)))
     (is (= ["1997" "Ford" "E350"] (second csv)))))
 
-(deftest writing-with-different-escape-char
+(deftest writing-quotes-with-different-escape-char
   (let [data [["This string \"contains\" quotes"]]
         writer (StringWriter.)]
     (write-csv writer data :escape \\)
     (is (= "\"This string \\\"contains\\\" quotes\"\n" (str writer)))))
 
-(deftest reading-with-different-escape-char
+(deftest writing-escape-char-with-different-escape-char
+  (let [data [["This string contains the escape char\\"]]
+        writer (StringWriter.)]
+    (write-csv writer data :escape \\)
+    (is (= "This string contains the escape char\\\n" (str writer)))))
+
+(deftest reading-quotes-with-different-escape-char
   (let [input "Key,Value\nExample,\"This \\\"contains\\\" quotes\""
         csv (read-csv input :escape \\)]
     (is (= 2 (count csv)))
     (is (= ["Key" "Value"] (first csv)))
     (is (= ["Example" "This \"contains\" quotes"] (second csv)))))
+
+(deftest reading-escape-char-with-different-escape-char
+  (let [input "Key,Value\nExample,\"This contains the escape char \\\\\""
+        csv (read-csv input :escape \\)]
+    (is (= 2 (count csv)))
+    (is (= ["Key" "Value"] (first csv)))
+    (is (= ["Example", "This contains the escape char \\"] (second csv)))))
+
+(defspec roundtripping 500
```

Comment:
Errors caused by escaping problems are moderately rare in the Elevate tests, but running the extract tests ~20 times was usually enough so have increased the test count here to 500 which should be enough. Could go for a round 1,000?

Reply:
(Is this the correct way to combine `defspec` and `are`?)

Reply:
Don't think so - try test.chuck for easier generators in a for-all something like (disclaimer, written in this text box, haven't checked if this compiles or works):
```
(require '[clojure.test.check.clojure-test :refer [defspec]]
         '[com.gfredericks.test.chuck.clojure-test :refer [for-all]])

(defspec positive-number-positivity-spec 100
  (for-all [headings (gen/vector gen/keyword)
                data (gen/vector (apply gen/hash-map (gen/interpose headings (repeatedly gen/string))))
                escape gen/char]
    (is (= data (roundtrip data escape)))
    (is (> x 0))))
```

Reply:
(char might be overkill, maybe ascii is sufficient, but if we *can* make it work... why not).

Also would be nice to generate all the different possible options for write-csv too, like line endings etc